File: GPL_Problems/LeetCode_Problem_657_Solution_1.cpp
class Solution {
public:
    bool judgeCircle(string moves) {

        //number of Ls = Rs and Us = Ds
        unordered_map<char, int> map;

        for(char c : moves)
        map[c]++;

        if(map['U'] == map['D'] && map['L'] == map['R'])
        return true;

        else
        return false;
        
    }
};

File: GPL_Problems/LeetCode_Problem_657_Solution_1.py
class Solution:
    def judgeCircle(self, moves: str) -> bool:
        l = 0
        r = 0
        u = 0
        d = 0

        for c in moves:
            if c == 'L':
                l += 1
            elif c == 'R':
                r += 1
            elif c == 'U':
                u += 1
            else:
                d += 1

        if l == r and u == d:
            return True
        else:
            return False